src/covid_model_seiir_pipeline/pipeline/forecasting/model/forecast_metrics.py

```diff
@@ -41,7 +57,29 @@ def compute_output_metrics(infection_data: pd.DataFrame,
     deaths = observed_deaths.combine_first(modeled_deaths)
     r_effective = compute_effective_r(components, seir_params, compartment_info.compartments)
 
-    return components, infections, deaths, r_effective
+    return OutputMetrics(
+        components=components, 
+        infections=infections, 
+        deaths=deaths, 
+        r_effective=r_effective
+    )
+
+
+def compute_corrected_hospital_usage(all_age_deaths: pd.DataFrame,
+                                     death_weights: pd.Series,
+                                     hospital_fatality_ratio: HospitalFatalityRatioData,
+                                     hospital_parameters: 'HospitalParameters',
+                                     correction_factors: HospitalCorrectionFactors) -> HospitalMetrics:
+    hospital_usage = compute_hospital_usage(
+        all_age_deaths.reset_index(),
+        death_weights,
+        hospital_fatality_ratio,
+        hospital_parameters,
+    )
+    hospital_usage.hospital_census = (hospital_usage.hospital_census * correction_factors.hospital_census).fillna(method='ffill')
```

Comment:
Just noting that I don't think the current hospital model does any imputation, it just drops values where the indices don't match.